### PR TITLE
Formatting tweaks, JSON fixes and tabs->spaces

### PR DESCRIPTION
**Pull request by ml-evs**
Just reading through the rendered spec on GitHub and found a couple of indents that are being treated as

> quoted text

where I don't think it was intentional.

Please check the rich diff to confirm this (the button is at the top right of the files changed window)

### CODE REVIEW

**Comment by ml-evs** [comment]
I think this is happening because we have some mixed tabs/spaces in the specification...

**Comment by CasperWA** [comment]
The diff also looks wrong now for code/JSON example sections.

**Comment by ml-evs** [comment]
Yep, trying to fix this properly with expandtab now

**Comment by ml-evs** [comment]
So I ended up going a bit overboard here, and have a script that can strip the JSON blocks out of the spec and validate them as JSON (a separate problem to the whitespace). This turned up lots of extra commas and missing quotes. I'm not sure if it's something that is worth adding to the CI itself as it is slightly manual (e.g. it skips our `//comments` so there are some false positives), but it has at least turned up some errors.

File: optimade.rst [hunk 1]
```diff
@@ -2165,8 +2165,8 @@ species
       Elements denoting vacancies MUST have masses equal to 0.
     - **original\_name**: OPTIONAL. Can be any valid Unicode string, and SHOULD contain (if specified) the name of the species that is used internally in the source database.
 
-          Note: With regards to "source database", we refer to the immediate source being queried via the OPTIMADE API implementation.
-	  The main use of this field is for source databases that use species names, containing characters that are not allowed (see description of the list property `species_at_sites`_).
+      Note: With regards to "source database", we refer to the immediate source being queried via the OPTIMADE API implementation.
```
Comment (CasperWA) [inline optimade.rst:2168]: Is this not supposed to be indented because it's a note? Maybe it's just confusing.

**Review by CasperWA — COMMENTED** [review]
Well ... this was *extremely* close to being a clean sheet... Also, it's just a comment/question :)

**Review by CasperWA — APPROVED** [review]
Bravo monsieur 👏 🥳

**Review by shyamd — APPROVED** [review]

**Review by CasperWA — APPROVED** [review]

**Review by shyamd — APPROVED** [review]

**Pull request merged**